feat: add informative tooltips to all fields

simboonlong opened the pull request:
feat: show alert tooltip when there is an endorsement awaiting holder's action
test: tooltip logic
chore: rename ownership to holdership
refactor: sec- out of ids

sumitnitsurat committed:
Added token owner information from oa-tone

sumitnitsurat committed:
test added for token owner detail view

sumitnitsurat committed:
reduce/saga for token ownership trasnfer added

sumitnitsurat committed:
color changed for transferable asset

sumitnitsurat committed:
fixed integration failed test

sumitnitsurat committed:
variable name fixed

sumitnitsurat committed:
variable name fixed

sumitnitsurat committed:
test added for detailed certioficate verify block

sumitnitsurat committed:
integration test fixed

sumitnitsurat committed:
Merge branch 'feat/token-interaction' into feat/token-transfer

simboonlong committed:
create TokenSideBar
basic styling of sidebar

simboonlong committed:
create TokenSideBar

simboonlong committed:
fix lint temporary

simboonlong committed:
fix lint temporary

simboonlong committed:
toggle sidebar
add surrender section

simboonlong committed:
toggle sidebar

simboonlong committed:
move tokensidebar into assetinfo folder

simboonlong committed:
move tokensidebar into assetinfo folder
separate sections based on transfer/change scenarios

simboonlong committed:
edit z-index of drawer so it will be behind sidebar in drawer
add hamburger to toggle sidebar

simboonlong committed:
edit z-index of drawer so it will be behind sidebar in drawer
add hamburger to toggle sidebar

simboonlong committed:
fix lint warnings

simboonlong committed:
fix lint warnings

simboonlong committed:
use .tsx instead
lint fix
css spacing tweaks

simboonlong committed:
use .tsx instead
lint fix
css spacing tweaks

sumitnitsurat committed:
updated jest config for module extension

sumitnitsurat committed:
updated jest config for module extension

sumitnitsurat committed:
Merge branch 'feat/token-siderbar' into feat/token-transfer

rjchow committed:
changed import string

rjchow committed:
changed import string

simboonlong committed:
refactor section into field component
TokenSideBarRole unit test
tweak box-shadow

simboonlong committed:
refactor section into field component
TokenSideBarRole unit test
tweak box-shadow

simboonlong committed:
Merge branch 'feat/token-siderbar' of github.com:TradeTrust/tradetrust-website into feat/token-siderbar

simboonlong committed:
standardise text casing

simboonlong committed:
standardise text casing

simboonlong committed:
Renamed tokenSideBar.scss to TokenSideBar.scss

simboonlong committed:
Renamed tokenSideBar.scss to TokenSideBar.scss

simboonlong committed:
fix addtional field logic in holder scenario

simboonlong committed:
fix addtional field logic in holder scenario

simboonlong committed:
rename TokenSideBarField casing

simboonlong committed:
rename TokenSideBarField casing

simboonlong committed:
lint fix

simboonlong committed:
lint fix

sumitnitsurat committed:
provider method added

sumitnitsurat committed:
Merge branch 'feat/ether-js' into feat/token-transfer

sumitnitsurat committed:
oa-token updated

sumitnitsurat committed:
test updated with shallow

simboonlong committed:
tokensidebar test

simboonlong committed:
tokensidebar test

simboonlong committed:
typo fix

simboonlong committed:
typo fix

sumitnitsurat committed:
tets for certificateVerifyBlock added

simboonlong committed:
Merge branch 'master' of github.com:TradeTrust/tradetrust-website into feat/token-usertype

sumitnitsurat committed:
Merge branch 'feat/token-siderbar' into feat/token-usertype

sumitnitsurat committed:
Merge branch 'feat/token-transfer' into feat/token-usertype

sumitnitsurat committed:
Merge branch 'feat/token-siderbar' into feat/token-transfer

sumitnitsurat committed:
Merge branch 'feat/token-interaction' into feat/token-transfer

simboonlong committed:
on hold

sumitnitsurat committed:
initliaze token added

simboonlong committed:
fix: to ignore .tsx test files on sonar

simboonlong committed:
fix: to ignore .tsx test files on sonar

sumitnitsurat committed:
Merge branch 'feat/token-usertype' of https://github.com/TradeTrust/opencerts-website into feat/token-usertype

sumitnitsurat committed:
reducer

sumitnitsurat committed:
added redux useSelectro amnd dispatch hooks to fetch data form demo services.

sumitnitsurat committed:
Merge branch 'master' into feat/token-usertype

sumitnitsurat committed:
AssetInfo text fixed

sumitnitsurat committed:
Merge branch 'master' into feat/token-sidebar

sumitnitsurat committed:
Merge branch 'feat/token-sidebar' into feat/token-usertype

sumitnitsurat committed:
temo

simboonlong committed:
refactor: breakdown TokenSideBar into smaller testable components
write more sensible test conditions
make edits based on comments

sumitnitsurat committed:
Merge branch 'feat/token-transfer' into feat/ether-js

sumitnitsurat committed:
Merge branch 'feat/token-sidebar' into feat/token-usertype

sumitnitsurat committed:
loading admin address and token initialization added

simboonlong committed:
refactor: get token addresses from reducers to determine which user view to render

simboonlong committed:
refactor: avoid duplicate code in TokenSideBarContent
rename import styles to css for consistency

sumitnitsurat committed:
[feat]: check if the token owner is escrow contract

simboonlong committed:
refactor: get UserRole by switch
test: check if sidebar content renders the correct view

sumitnitsurat committed:
type for provider added

sumitnitsurat committed:
Merge remote-tracking branch 'origin/feat/token-userview' into feat/token-transfer

simboonlong committed:
fix: tests
fix: only show Surrender Document when adminAddr === beneAddr && where adminAddr === holderAddr
fix: only show Approve Change Beneficiary when there is endorsement and in Holder view

sumitnitsurat committed:
transfer holdership

simboonlong committed:
feat: show no address match view
fix: text change from transfer ownership > transfer holdership

sumitnitsurat committed:
typing fixed

simboonlong committed:
fix: incorrect condition of holder + bene without pending endorsement
test: more specific tests on TokenSideBar
feat: show a spinner loader when application is fetching token response.
feat: show a default no access message when token registry does not match with user token.

sumitnitsurat committed:
ignore integration test from coverage

simboonlong committed:
test(token sidebar reducer): tests on responses when requesting user token (#51)

* test(token sidebar reducer): tests on responses when requesting user token

* test: handle get token addresses saga test cases

sumitnitsurat committed:
readonly token instance fixed

sumitnitsurat committed:
Merge remote-tracking branch 'origin/feat/token-sidebar' into feat/token-all-feat

sumitnitsurat committed:
Merge branch 'feat/token-userview' into feat/token-all-feat

sumitnitsurat committed:
Merge branch 'feat/token-usertype' into feat/token-all-feat

sumitnitsurat committed:
Merge branch 'feat/token-transfer' into feat/token-all-feat

sumitnitsurat committed:
Merge branch 'fix/token-service-instance' into feat/token-all-feat

sumitnitsurat committed:
lint fixed

sumitnitsurat committed:
chekc escrow refactored

sumitnitsurat committed:
refactored loading admin address

simboonlong committed:
feat: add informative tooltips to all fields
feat: show alert tooltip when there is an endorsement awaiting holder's action
test: tooltip logic
chore: rename ownership to holdership
refactor: sec- out of ids

simboonlong committed:
chore: stylelint fix, token test

sumitnitsurat commented on src/components/AssetInfo/TokenSideBar.test.tsx:
```diff
@@ -32,8 +32,8 @@ describe("tokenSideBar", () => {
-    expect(wrapper.find("#sec-transferownership h4").text()).toStrictEqual("Transfer Holdership");
-    expect(wrapper.find("#sec-transferownership button").text()).toStrictEqual("Transfer");
+    expect(wrapper.find("#sec-transferholdership h4").text()).toStrictEqual("Transfer Holdership");
+    expect(wrapper.find("#sec-transferholdership button").text()).toStrictEqual("Transfer");
```
why this "sec-" in ID ?

sumitnitsurat commented on src/components/AssetInfo/TokenSideBarTooltip.tsx:
```diff
@@ -0,0 +1,93 @@
+            effect="solid"
+            getContent={() => {
+              return <p>Endorsement awaiting</p>;
+            }}
```
getContent=(() => (<p> Endorsement awaiting</p>));

sumitnitsurat commented on src/components/AssetInfo/TokenSideBarTooltip.tsx:
```diff
@@ -0,0 +1,93 @@
+
+const TokenSideBarTooltip = ({ id, approvedBeneficiaryAddress }: TokenSideBarTooltipProps) => {
+  const getTooltipContent = () => {
+    switch (true) {
```
Try to refactor it as discussed.

simboonlong committed:
refactor: switch case, getContent
test: check for 2 tooltip icons for Change Beneficiary with awaiting endorsement

simboonlong committed:
fix: spacing with h4

simboonlong commented:
abandoning this branch, will be redo-ing the feature at https://github.com/TradeTrust/tradetrust-website/pull/67
